drivers/display/display_renesas_lcdc.c

```diff
@@ -0,0 +1,595 @@
+	k_sem_init(&data->device_sem, 1, 1);
+	/* Event should be signaled by LCDC ISR */
+	k_sem_init(&data->sync_sem, 0, 1);
+	/* Event should be signaled by DMA ISR */
+	k_sem_init(&data->dma_sync_sem, 0, 1);
+
+	/* As per docs, display port should be enabled by default. */
+	if (gpio_is_ready_dt(&config->disp)) {
+		ret = gpio_pin_configure_dt(&config->disp, GPIO_OUTPUT_ACTIVE);
+		if (ret < 0) {
+			LOG_ERR("Could not activate display port");
+			return -EIO;
+		}
+	}
+
+	data->current_pixel_size = DISPLAY_SMARTBOND_PIXEL_SIZE;
+	data->current_pixel_format = DT_INST_PROP(0, pixel_format);
+
+	ret = display_smartbond_resume(dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = display_smartbond_dma_config(dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	IRQ_CONNECT(SMARTBOND_IRQN, SMARTBOND_IRQ_PRIO, smartbond_display_isr,
+								DEVICE_DT_INST_GET(0), 0);
+
+#if CONFIG_PM
+	/*
+	 * When in continues mode, the display device should always be refreshed
+	 * and so the controller is not allowed to be turned off. The latter, is
+	 * powered by PD_SYS which is turned off when the SoC enters the extended
+	 * sleep state. By acquiring PD_SYS, the deep sleep state is prevented
+	 * and the system enters the low-power state (i.e. ARM WFI) when possible.
+	 *
+	 * XXX CONFIG_PM_DEVICE_RUNTIME is no supported yet!
+	 */
+	da1469x_pd_acquire_noconf(MCU_PD_DOMAIN_SYS);
+#endif
+
+	return 0;
+}
+
+static int display_smartbond_blanking_on(const struct device *dev)
+{
+	const struct display_smartbond_config *config = dev->config;
+	struct display_smartbond_data *data = dev->data;
+	int ret = 0;
+
+	k_sem_take(&data->device_sem, K_FOREVER);
+
+	/*
+	 * This bit will force LCD controller's output to blank that is,
+	 * the controller will keep operating without outputting any
+	 * pixel data.
+	 */
+	LCDC->LCDC_MODE_REG |= LCDC_LCDC_MODE_REG_LCDC_FORCE_BLANK_Msk;
+
+	/* If enabled, disable display port. */
+	if (gpio_is_ready_dt(&config->disp)) {
+		ret = gpio_pin_configure_dt(&config->disp, GPIO_OUTPUT_INACTIVE);
+		if (ret < 0) {
+			LOG_WRN("Display port could not be de-activated");
+		}
+	}
+
+	k_sem_give(&data->device_sem);
+
+	return ret;
+}
+
+static int display_smartbond_blanking_off(const struct device *dev)
+{
+	const struct display_smartbond_config *config = dev->config;
+	struct display_smartbond_data *data = dev->data;
+	int ret = 0;
+
+	k_sem_take(&data->device_sem, K_FOREVER);
+
+	/* If used, enable display port */
+	if (gpio_is_ready_dt(&config->disp)) {
+		ret = gpio_pin_configure_dt(&config->disp, GPIO_OUTPUT_ACTIVE);
+		if (ret < 0) {
+			LOG_WRN("Display port could not be activated");
+		}
+	}
+
+	/*
+	 * This bit will force LCD controller's output to blank that is,
+	 * the controller will keep operating without outputting any
+	 * pixel data.
+	 */
+	LCDC->LCDC_MODE_REG &= ~LCDC_LCDC_MODE_REG_LCDC_FORCE_BLANK_Msk;
+
+	k_sem_give(&data->device_sem);
+
+	return ret;
+}
+
+static int display_smartbond_set_orientation(const struct device *dev,
```

Comment:
Is this implementation needed? The display API no longer requires all functions be implemented, it will return `-ENOSYS` if one isn't present.